.github/workflows/serverless-service.yml

```diff
@@ -3,80 +3,113 @@
 name: Serverless Service CI/CD
 
 permissions:
-  id-token: write # This is required for requesting the JWT
-  issues: write
-  checks: read
   contents: read
 
+env:
+  NODE_VERSION: "18"
+  PYTHON_VERSION: "3.11"
+  AWS_REGION: "us-east-1"
+
 on:
   push:
     branches: [main]
+    paths:
+      - "product/**"                # source code
+      - "infrastructure/**"         # CDK source code
+      - "tests/**"
```

Comment:
also changed in .github

Reply:
You mean add this workflow as a trigger if ever changed? Otherwise you'll trigger this workflow even with a change in `PULL_REQUEST_TEMPLATE.md`, etc.

Ideally, we'd break this workflow into smaller ones too -- one that you run quality gates, one for slower jobs like integ/e2e, one for linting workflows, etc.

Lemme know if I understood it and I can push the change